lib/bundler/plugin.rb

```diff
@@ -166,9 +197,11 @@ def validate_plugin!(plugin_path)
     def register_plugin(name, spec, optional_plugin = false)
       commands = @commands
       sources = @sources
+      hooks = @hooks_by_event
 
       @commands = {}
       @sources = {}
+      @hooks_by_event = Hash.new {|h, k| h[k] = [] }
```

Comment:
`reset!` also removes all other instance variables along with `@loaded_plugin_names` .  Calling `reset!` may have undesirable side effects.

Also I don't think this list will expand where as reset may expand.